iphone/Classes/TiCalendarCalendar.m

```diff
@@ -291,6 +291,37 @@ -(NSString*)name
     return [[self calendar] title];
 }
 
+-(NSString*)sourceTitle
+{
+    if (![NSThread isMainThread]) {
+        __block id result;
+        TiThreadPerformOnMainThread(^{result = [[[self calendar] source] title];}, YES);
+        return [result autorelease];
+    }
+    return [[[self calendar] source] title];
```

Comment:
Why not just:
```objc
 __block id result;

TiThreadPerformOnMainThread(^{
    result = [[[self calendar] source] title];
}, YES);

return result;
```
If we are already on main-thread, `TiThreadPerformOnMainThread` will take of it and dispatch the block on it. Same for all other methods here.